daemon/images/images.go

```diff
@@ -237,15 +239,15 @@ func (i *ImageService) Images(imageFilters filters.Args, all bool, withExtraAttr
 				}
 
 				if layerRefs[chid] > 1 {
```

Comment:
If I see it correctly, it looks like `diffSize` is only used in this case, so we could put the `diffSize, err := allLayers[chid].DiffSize()` code inside this diff

On a side-note: looks like none of the `DiffSize()` implementations actually return an error 🤔. Not sure why it was added (error return); perhaps for "consistency")